shapely/strtree.py

```diff
@@ -389,25 +401,86 @@ def nearest_all(self, geometry, max_distance=None, return_distance=False):
 
         Examples
         --------
-        >>> import shapely
-        >>> tree = shapely.STRtree(shapely.points(np.arange(10), np.arange(10)))
-        >>> tree.nearest_all(shapely.points(1,1)).tolist()  # doctest: +SKIP
-        [[0], [1]]
-        >>> tree.nearest_all( shapely.box(1,1,3,3)]).tolist()  # doctest: +SKIP
-        [[0, 0, 0], [1, 2, 3]]
-        >>> points = shapely.points(0.5,0.5)
-        >>> index, distance = tree.nearest_all(points, return_distance=True)  # doctest: +SKIP
-        >>> index.tolist()  # doctest: +SKIP
-        [[0, 0], [0, 1]]
-        >>> distance.round(4).tolist()  # doctest: +SKIP
+        >>> import numpy as np
+        >>> from shapely import box, Point
+        >>> points = [Point(0, 0), Point(1, 1), Point(2,2), Point(3, 3)]
+        >>> tree = STRtree(points)
+
+        Find the nearest tree geometries to a scalar geometry:
+
+        >>> indices = tree.query_nearest(Point(0.25, 0.25))
+        >>> indices.tolist()
+        [0]
+
+        Retrieve the tree geometries by results of query:
+
+        >>> tree.geometries.take(indices).tolist()
+        [<shapely.Point POINT (0 0)>]
+
+        Find the nearest tree geometries to an array of geometries:
+
+        >>> query_points = np.array([Point(1, 1), Point(2.25, 2.25)])
+        >>> arr_indices = tree.query_nearest(query_points)
+        >>> arr_indices.tolist()
+        [[0, 1], [1, 2]]
+
+        Or transpose to get all pairs of input and tree indices:
+
+        >>> arr_indices.T.tolist()
+        [[0, 1], [1, 2]]
+
+        Retrieve all pairs of input and tree geometries:
+
+        >>> np.array([query_points.take(arr_indices[0]), tree.geometries.take(arr_indices[1])]).T.tolist()
+        [[<shapely.Point POINT (1 1)>, <shapely.Point POINT (1 1)>],
+         [<shapely.Point POINT (2.25 2.25)>, <shapely.Point POINT (2 2)>]]
+
+        All intersecting geometries in the tree are returned by default:
+
+        >>> tree.query_nearest(box(1,1,3,3)).tolist()
+        [1, 2, 3]
+
+        Set all_matches to False to to return a single match per input geometry:
+
+        >>> tree.query_nearest(box(1,1,3,3), all_matches=False).tolist()  # doctest: +SKIP
```

Comment:
I removed most of the `# doctest: +SKIP` now that the doctest infrastructure is using GEOS > 3.6.  This is retained here becuse the output is nondeterministic between GEOS versions; the output may be either `[0]` or `[1]`; both are equally valid.